docs/references.rst

```diff
@@ -184,12 +184,59 @@ keep making your query bigger and bigger::
             ['total_gross', 'aggregate'=>'sum'],
         ]);
 
-.. important::
-    Imported fields will preserve format of the field the reference. In the example,
-    if 'Invoice_line' field total_vat has type `money` then it will also be used
-    for a sum. Aggregate fields are always declared read-only, and if you try to
-    change them, you will receive exception.
+Imported fields will preserve format of the field they reference. In the example,
+if 'Invoice_line' field total_vat has type `money` then it will also be used
+for a sum. 
+
+You can also specify a type yourself::
+
+    ->addField('paid_amount', ['aggregate'=>'sum', 'field'=>'amount', 'type'=>'money']);
+
+Aggregate fields are always declared read-only, and if you try to
+change them (`$m['paid_amount'] = 123;`), you will receive exception.
+
+Available Aggregation Functions
+-------------------------------
+
+The mathematical aggregates such as sum, avg, min, max and count will automatically
+default to 0 if no respective rows were provided. The default SQL behaviour is to
```

Comment:
Not sure if this is good or bad to use `coalesce` for all of these.
`count` - surely should be 0 not NULL
`sum` - probably also 0 is more meaningful than NULL
`min`, `max` and `avg`- 0 will give wrong assumption that actual value is 0. Not so sure if this is ok. Sometimes 0 is ok here, but i think there can be use-cases where we would like to see NULL to differ actual value (like 0) from no value at all (like NULL).

Reply:
ok, i agree, i'll switch the other methods to use 'fx' not 'fx0'

Reply:
ok